external/blobManager.ts

```diff
@@ -0,0 +1,54 @@
+import { ContainerURL } from '@azure/storage-blob';
+import {
+    Configuration,
+    IItemRecordsBlob,
+} from 'cloud-docs-shared-code';
+import { ReferenceOperation } from 'cloud-docs-shared-code/reference/preprocessedModels';
+
+const BlobStorage = require('@azure/storage-blob');
+
+export const storeReferenceDataToBlobStorage = async (
+    dataBlob: IItemRecordsBlob,
+    operation: ReferenceOperation,
+): Promise<void> => {
+    const containerUrl = getContainerUrl();
+    const blobName = getBlobName(dataBlob.id, operation);
+    const blobURL = BlobStorage.BlockBlobURL.fromContainerURL(containerUrl, blobName);
+
+    const blob = JSON.stringify(dataBlob);
+
+    await blobURL.upload(
+        BlobStorage.Aborter.none,
+        blob,
+        blob.length,
+    );
+};
+
+const getContainerUrl = (): ContainerURL => {
+    const sharedKeyCredential = new BlobStorage.SharedKeyCredential(
+        Configuration.keys.azureAccountName,
+        Configuration.keys.azureStorageKey,
+    );
+    const pipeline = BlobStorage.StorageURL.newPipeline(sharedKeyCredential);
+    const serviceUrl = new BlobStorage.ServiceURL(
+        `https://${Configuration.keys.azureAccountName}.blob.core.windows.net`,
+        pipeline,
+    );
+
+    return BlobStorage.ContainerURL.fromServiceURL(serviceUrl, Configuration.keys.azureContainerName);
+};
+
+export const getBlobName = (id: string, operation: ReferenceOperation): string => {
+    switch (operation) {
+        case ReferenceOperation.Update:
+        case ReferenceOperation.Initialize: {
+            return id;
```

Comment:
Why doesn't this use the ".json" extension as well?

Reply:
Great point. At first, we wanted to be consistent with the current implementation of Tutorial Search, which doesn't store blobs with the ".json" extension. However, I have missed out that here in the case of a `ReferenceOperation.Preview`, the blob does get the extension here anyways. I guess it will be clearer to have the extensions there every time, therefore I will add them to Tutorial Search as well.